internal/subscriber/retry/delayconsumer_test.go

```diff
@@ -81,6 +61,148 @@ func TestDelayConsumer_Run(t *testing.T) {
+	mockBrokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+	mockConsumer.EXPECT().Close(gomock.Any()).Return(nil)
+	tests := []struct {
+		name    string
+		wantErr bool
+	}{
+		{
+			name:    "Test Delay Consumer one message counsume out of 2",
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			go dc.Run(ctx)
+			select {
+			case err := <-errCh:
+				if !tt.wantErr {
+					t.Errorf("Got Error : %v", err)
+				}
+			case <-time.NewTicker(time.Second * 2).C:
+				assert.Equal(t, len(dc.cachedMsgs), 1)
+				cancel()
+				<-time.NewTicker(time.Second * 1).C
+			}
+		})
+	}
+}
+
+func TestDelayConsumer_deadLetter(t *testing.T) {
+	ctx := context.Background()
+	subCtx, cancel := context.WithCancel(ctx)
+	ctrl := gomock.NewController(t)
+	mockBrokerStore := mocks.NewMockIBrokerStore(ctrl)
+	mockHandler := mocks3.NewMockMessageHandler(ctrl)
+	mockConsumer := mocks2.NewMockConsumer(ctrl)
+	cache := cachemock.NewMockICache(ctrl)
+
+	subs := getDummySubscriptionModel()
+	subscriberID := "subscriber-dl-id"
+	mockBrokerStore.EXPECT().GetConsumer(gomock.Any(), gomock.Any()).Return(mockConsumer, nil)
+	mockConsumer.EXPECT().Resume(gomock.AssignableToTypeOf(subCtx), gomock.Any()).Return(nil).AnyTimes()
+	mockConsumer.EXPECT().Pause(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	// initialize delay consumer
+	errCh := make(chan error)
+	dc, err := NewDelayConsumer(subCtx, subscriberID, "t1-dl", subs, mockBrokerStore, mockHandler, cache, errCh)
+	producer := buildMockProducer(ctx, ctrl)
+	assert.NotNil(t, dc)
+	assert.Nil(t, err)
+	assert.NotNil(t, dc.LogFields())
+	tFut := time.Now().Add(time.Second * 10)
+	msgs := make([]messagebroker.ReceivedMessage, 0)
+	msg := getDummyBrokerMessageDLQ()
+	msg.NextDeliveryTime = tFut
+	msgs = append(msgs, msg)
+	resp := &messagebroker.GetMessagesFromTopicResponse{Messages: msgs}
+	count := 0
+	mockConsumer.EXPECT().ReceiveMessages(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+			if count == 0 {
+				count++
+				return resp, nil
+			} else {
+				return &messagebroker.GetMessagesFromTopicResponse{}, nil
+			}
+		}).AnyTimes()
+	cache.EXPECT().Get(gomock.Any(), gomock.Any()).Return([]byte{'0'}, nil).AnyTimes()
+	cache.EXPECT().Set(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	// on new message from broker
+	mockHandler.EXPECT().Do(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	mockConsumer.EXPECT().CommitByPartitionAndOffset(gomock.Any(), gomock.Any()).Return(messagebroker.CommitOnTopicResponse{}, nil).AnyTimes()
+	mockBrokerStore.EXPECT().RemoveConsumer(gomock.Any(), gomock.Any()).Return(true)
+	mockBrokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+	mockConsumer.EXPECT().Close(gomock.Any()).Return(nil)
```

Comment:
there are so many expect calls here.. lets see if we can clean this up by moving these to a separate setup kind of func.

Reply:
moved to setup